XMonad/Hooks/DynamicLog.hs

```diff
@@ -746,6 +764,196 @@ xmobarStripTags tags = strip [] where
+-- boilerplate is reduced.
+-- If you wish to have multiple status bars managed by xmonad, you can do so by creating
+-- the status bar configs and combining them with 'combineSBs':
+--
+-- > xmobarTop    = statusBarHandleConfig "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_top"    ppTop
+-- > xmobarBottom = statusBarHandleConfig "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_bottom" ppBottom
+-- > xmobar1      = statusBarHandleConfig "xmobar -x 1 $HOME/.config/xmobar/xmobarrc1"       pp1
+-- > main = xmonad =<< combineSBs [xmobarTop, xmobarBottom, xmobar1] myConfig
+--
+-- Or if you're feeling adventurous:
+--
+-- > myBars = map (uncurry statusBarHandleConfig) [ ("xmobar -x 0 $HOME/.config/xmobar/xmobarrc_top",    ppTop)
+-- >                                              , ("xmobar -x 0 $HOME/.config/xmobar/xmobarrc_bottom", ppBottom)
+-- >                                              , ("xmobar -x 1 $HOME/.config/xmobar/xmobarrc1",       pp1) ]
+-- > main = xmonad =<< combineSBs myBars myConfig
+--
+-- To change the default keybinding or action to toggle struts, you can do the following:
+--
+-- > ...
+-- > mySBs = withAction myAction . combineSBConfs <$> sequence myBars
+-- > main = xmonad =<< makeStatusBar myConfig =<< mySBs
+--
+-- if you also want to change the key, you can chain the transformations like so:
+--
+-- > ...
+-- > mySBCombinator = withKey myToggleStrutsKey . withAction myAction . combineSBConfs <$> sequence myBars
+-- > ...
+--
+-- The above examples also work if the different status bars support different
+-- logging methods: you could do mix property logging and logging via standard input.
+-- One thing to keep in mind: if multiple bars read from the same property, their content
+-- will be the same. If you want to use property-based logging with multiple bars,
+-- they should read from different properties.
+
+-- | Reduces a list of (pure) 'StatusBarConfig's into a single one, by sequencing
+-- their respective hooks. To avoid ambiguitiy, the key used to toggle struts and
+-- the action performed by it are the defaults. If multiple status bars with different
+-- properties are given, make sure they read from different properties.
+-- To directly combine confs created by 'statusBarHandleConfig', 'statusBarHandleConfig''
+-- or 'statusBarPropConfig', take a look at 'combineSBs' instead,
+-- which takes instead a list of 'StatusBarConfig's that are wrapped in 'IO'.
+combineSBConfs :: [StatusBarConfig] -- ^ The list of configs to be merged
+               -> StatusBarConfig
+combineSBConfs confs = def
+  { sbLogHook     = mapM_ sbLogHook confs
+  , sbStartupHook = mapM_ sbStartupHook confs
+  , sbCleanupHook = mapM_ sbCleanupHook confs
+  }
+
+-- | Like 'combineSBConfs', but expects the status bars to be wrapped in 'IO', and applies
+-- the combined status bars to the base config. This is useful to directly combine configs
+-- returned by 'statusBarHandleConfig', 'statusBarHandleConfig'' and 'statusBarPropConfig'.
```

Comment:
It's also worth noting that if one chooses not to "directly combine configs" and instead uses `makeStatusBar` multiple times, the `ch1 *> sh1 *> ch2 *> sh2 *> …` sequence will result in killing of all but the last status bar.

Also, `AvoidStruts` will be applied to the layout multiple times, and eventhooks will run multiple times.

This is probably worth mentioning somewhere. But I'd postpone this until we deal with my other comments which may or may not results in changes to the API.

Reply:
That's a really good point, it slipped my mind to mention that in the docs

Reply:
We could drop `makeSBwithKey` if we decide to only leave `statusBar` and remove the other variants

Reply:
But I think `makeSBwithKey` isn't too bad. It makes it easy for users to customize the bars a bit without touching the keys/layout plumbing. :-)

Reply:
Since the naming is confusing and `makeSBwithKey` does more than just the key, I think it needs another name (maybe `makeStatusBar` and `makeStatusBar'`